### PR TITLE
[backend] implement model- teamModel

### PR DESCRIPTION
**Pull request by bontu-fufa**

### CODE REVIEW

File: Backend/models/teamModel.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+var mongoose = require('mongoose')
+var teamSchema = mongoose.Schema({
+    _id: mongoose.Schema.Types.ObjectId,
+    name: {
+        type: String,
+        required: true,
+    },
+    dob: {
```
Comment (Infinidrix) [inline Backend/models/teamModel.js:8]: Should it be short like dob or like date_of_birth and dateOfBirth? I think being descriptive is good :)

File: Backend/models/teamModel.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    description : {
+        type: String,
+        required: true
+    },
+    country: {
+        type: String,
+        required: true
+    },
+})
+const Team = module.exports = mongoose.model('Team', teamSchema)
```
Comment (Infinidrix) [inline Backend/models/teamModel.js:33]: I don't know why there's a `const Team =`  on this line, maybe we could do it like this:
```
const Team = mongoose.model('Team', teamSchema);
module.exports = Team;
```

**Review by Infinidrix — COMMENTED** [review]
Some comments. Also Yisehak and Beimnet are discussing about the model in the doc so let's not merge it yet

File: Backend/models/teamModel.js (hunk 2 above)
Comment (bontu-fufa) [reply Backend/models/teamModel.js:33]: Actually, it's the same it works in both ways.
yours is more readable.

File: Backend/models/teamModel.js (hunk 1 above)
Comment (bontu-fufa) [reply Backend/models/teamModel.js:8]: okay okay

Comment (bontu-fufa) [reply Backend/models/teamModel.js:8]: let's use date_of_birth

Comment (Infinidrix) [reply Backend/models/teamModel.js:8]: That sounds great

File: Backend/models/teamModel.js (hunk 2 above)
Comment (Infinidrix) [reply Backend/models/teamModel.js:33]: Yeah it's the same thing essentially

**Review by Infinidrix — APPROVED** [review]

File: Backend/controllers/TeamController.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    const team = await Team.findById(req.params.id);
+    return res.send(team);
+};
+
+
+// Post a team member
+exports.post_member_data = async (req, res) => {
+    const team = new Team(req.body);
+    
+    const notValid = teamValidator(team);
```
Comment (Khalid-Sultan) [inline Backend/controllers/TeamController.js:41]: To be uniform in all cases, I think using Antii's system wide validation techniques would be better. Like adding the middle  ware on the routes and letting them handle it to make the main code a bit leaner. For reference look into how the validators are implemented in the Information model, routes and controllers

File: Backend/controllers/TeamController.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+// update a team
+exports.update_team = async (req, res) => {
+    let team = await Team.findById(req.params.id);
+    if (!team) {
+        return res.status(404).send("team member doesn't exist.");
+    }
+
+    team.set(req.body)
+
+    const notValid = teamValidator(team);
```
Comment (Khalid-Sultan) [inline Backend/controllers/TeamController.js:62]: Same goes for the validation here as well

**Review by Khalid-Sultan — APPROVED** [review]
LGTM :star:

**Pull request closed without merging**